Makefile

```diff
@@ -296,10 +296,18 @@ dist: doc
 	(cd $(TMP)/lxd-$(VERSION)/vendor/liblxc ; git show-ref HEAD | cut -d' ' -f1 > .gitref)
 
 	# Copy doc output
-	cp -r doc/_build $(TMP)/lxd-$(VERSION)/doc/html/
+	cp -r --preserve=mode doc/_build $(TMP)/lxd-$(VERSION)/doc/html/
 
 	# Assemble tarball
-	tar --exclude-vcs -C $(TMP) -zcf $(ARCHIVE).gz lxd-$(VERSION)/
+	$(eval SOURCE_EPOCH := $(shell TZ=UTC0 git log -1 --format=tformat:%cd --date=format:%Y-%m-%dT%H:%M:%SZ $(HASH)))
```

Comment:
This is very complicated and not clear what these lines are doing and why (beyond the stated goal in the commit message). Please can you add some comments to thd file explaining what we are doing here and why for future readers. Thanks

Reply:
I meant to include a link to https://www.gnu.org/software/tar/manual/html_node/Reproducibility.html but this was lost during a rebase it seems. I added it back along with more explanation as that is indeed rather cryptic.